### ISSUE TITLE
Automate maven central publishing

### ISSUE THREAD

**Issue by steffen-wilke** [gurkenlabs/litiengine#439]
Right now we publish our library manually from my local machine to Maven Central. This process is tedious and requires repetitive work to be done which also makes it error-prone and it relies on my local setup.

Also, the entire process of publishing the engine is currently described in a private repo which needs to be kept up-to-date. Automating the process would add the process as code which increases reliability.

I recently found this gem that describes how to automate this process with GitHub Actions:
[How to Sign and Release to The Central Repository with GitHub Actions](https://gist.github.com/sualeh/ae78dc16123899d7942bc38baba5203c)

In short, we'll need to do the following:

1. Store the secrets as GitHub secrets
2. Add a new workflow that runs the publishing process (including the singing of the artifacts) 
   * Triggered manually by one of the repository owners
   * Provide an input to set the `release = true` flag

**Additional thoughts**
This is somewhat related to #438 and #436

**Comment by steffen-wilke** [comment]
I've added the required secrets to our repository:

- `OSSRH_GPG_SECRET_KEY`
- `OSSRH_GPG_SECRET_KEY_PASSWORD`
- `OSSRH_USERNAME`
- `OSSRH_PASSWORD`

**Comment by weisJ** [comment]
For the gpg key there is an inherent issue with passing it through GitHub actions because it has multiple lines. My solution for this problem was to save the key with `#` instead of new line characters and replace them in the build script:
https://github.com/weisJ/darklaf/blob/378b6f19eae5f5e7b8ac997ea20490c3edd6776d/build.gradle.kts#L213

I’ll adjust #436 to add support publishing to maven central. Probably there should be a separate action for snapshots and releases. Both should probably create a corresponding release + tag in the repo which included the jpackage artifacts.

**Comment by steffen-wilke** [comment]
Apparently replacing newlines with `\n` also does the trick (see [How to Sign and Release to The Central Repository with GitHub Actions](https://gist.github.com/sualeh/ae78dc16123899d7942bc38baba5203c)).

I did that for the `OSSRH_GPG_SECRET_KEY` to make it a one-line value.

**Comment by steffen-wilke** [comment]
@weisJ Since I'm currently fiddling around with this:

The publishing plugin auto-generates tasks based on the `PubName` (see [Maven Publish Plugin Docs](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:tasks)), which is something I'm currently struggling with. The actual `publish` task currently does nothing and right now I'm only able to publish to `mavenLocal` but there are no tasks for the `staging` repo.

**Comment by steffen-wilke** [comment]
@weisJ Thank you very much! The new version has been published to Maven Central.

Do you know whether it's possible to create the Github release just as a draft? We like to provide some more manual info there about the release and what it contains. Triggering a release also notifies followers of the repository but without the adjusted Release Notes ("just" the auto-generated) one.

**Comment by weisJ** [comment]
Yes simply add `draft: true` to the action in the workflow :)
